applications/FluidDynamicsApplication/python_scripts/apply_mass_conservation_check_process.py

```diff
@@ -54,12 +81,98 @@ def ExecuteBeforeSolutionLoop(self):
     def ExecuteInitializeSolutionStep(self):
         self.mass_conservation_check_process.ExecuteInitializeSolutionStep()
 
+    def Execute(self):
+        self.ExecuteFinalizeSolutionStep()
 
     def ExecuteFinalizeSolutionStep(self):
 
-        log_line_string = self.mass_conservation_check_process.ExecuteInTimeStep()
+        ### always necessary to keep track of the balanced volume
+        log_line_string = self.mass_conservation_check_process.ComputeBalancedVolume()
 
-        # writing first line in file
+        ### writing first line in file
         if ( self._write_to_log and self._is_printing_rank ):
             with open(self._my_log_file, "a+") as logFile:
-                logFile.write( log_line_string )
\ No newline at end of file
+                logFile.write( log_line_string )
+
+        # (1) #
+        if ( self._perform_local_corr ):
+            ### perform the local conservation procedure
+            # REMARK: The pseudo time step dt has no meaning as a physical time step and is NOT accounted as such.
+            # Instead, it as purely used for an artificial convection that helps to conserve the mass.
+            # Accordingly, dt depends on the currently necessary mass convection.
+            dt = self.mass_conservation_check_process.ComputeDtForConvection()
+
+            # REMARK: The distance field is artificially convected to extrapolate the inteface motion into the future.
+            # The time step considered for this artificial "forward convection" is dt.
+            # The previous distance field is saved in an auxiliary variable
+            prev_dt = self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
+            self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME] = dt
+            KratosMultiphysics.VariableUtils().SaveScalarVar(
+                KratosMultiphysics.DISTANCE,
+                KratosFluid.AUX_DISTANCE,
+                self._fluid_model_part.GetCommunicator().LocalMesh().Nodes)
+            self.forward_convection_process.Execute()
+            self._fluid_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME] = prev_dt
```

Comment:
These lines 108-115 are the main difference with respect to the first version. Distance before convection is stored in AUX_DISTANCE and then compared with the convected DISTANCE. Also, delta time is changed here to solve the convection step and then, the previous value is recovered.